client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,580 @@
+				if !ok {
+					return invalidTypeErr
+				}
+				if len(snapsHeader) == 0 {
+					// unexpected why this is an empty list, but just ignore for
+					// now
+					continue
+				}
+				fmt.Fprintf(w, "snaps:\n")
+				for _, sn := range snapsHeader {
+					snMap, ok := sn.(map[string]interface{})
+					if !ok {
+						return invalidTypeErr
+					}
+					// iterate over all keys in the map in a stable, visually
+					// appealing ordering
+					// first do snap name, which will always be present since we
+					// parsed a valid assertion
+					name := snMap["name"].(string)
+					fmt.Fprintf(w, "  - name:\t%s\n", name)
+
+					// the rest of these may be absent, but they are all still
+					// simple strings
+					for _, snKey := range []string{"id", "type", "default-channel", "presence"} {
+						snValue, ok := snMap[snKey]
+						if !ok {
+							continue
+						}
+						snStrValue, ok := snValue.(string)
+						if !ok {
+							return invalidTypeErr
+						}
+						if snStrValue != "" {
+							fmt.Fprintf(w, "    %s:\t%s\n", snKey, snStrValue)
+						}
+					}
+
+					// finally handle "modes" which is a list
+					modes, ok := snMap["modes"]
+					if !ok {
+						continue
+					}
+					modesSlice, ok := modes.([]interface{})
+					if !ok {
+						return invalidTypeErr
+					}
+					if len(modesSlice) == 0 {
+						continue
+					}
+
+					modeStrSlice := make([]string, 0, len(modesSlice))
+					for _, mode := range modesSlice {
+						modeStr, ok := mode.(string)
+						if !ok {
+							return invalidTypeErr
+						}
+						modeStrSlice = append(modeStrSlice, modeStr)
+					}
+					modesSliceYamlStr := "[" + strings.Join(modeStrSlice, ", ") + "]"
+					fmt.Fprintf(w, "    modes:\t%s\n", modesSliceYamlStr)
+				}
+
+			// long base64 key we can rewrap safely
+			case "device-key":
+				headerString, ok := headerValue.(string)
+				if !ok {
+					return invalidTypeErr
+				}
+				// the string value here has newlines inserted as part of the
+				// raw assertion, but base64 doesn't care about whitespace, so
+				// it's safe to split by newlines and re-wrap to make it
+				// prettier
+				headerString = strings.Join(
+					strings.Split(headerString, "\n"),
+					"")
+				fmt.Fprintln(w, "device-key: |")
+				strutil.WordWrapPadded(w, []rune(headerString), "  ", termWidth)
+
+			// the default is all the rest of short scalar values, which all
+			// should be strings
+			default:
+				headerString, ok := headerValue.(string)
+				if !ok {
+					return invalidTypeErr
+				}
+				fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+			}
+		}
+	}
+
+	return w.Flush()
+}
+
+// PrintModelAssertionJSON will format the provided serial or model assertion based on the parameters given in
+// JSON format. The output will be written to the provided io.Writer.
+func PrintModelAssertionJSON(w *tabwriter.Writer, modelFormatter ModelFormatter, useSerial, absTime, assertion bool, modelAssertion *asserts.Model, serialAssertion *asserts.Serial) error {
```

Comment:
same comments, I think this should have two variants PrintModelAssertionJSON and PrintSerialAssertionJSON,  we don't need the 2nd one immediatly

Reply:
I'll get this updated